drivers/memc/memc_smartbond_nor_psram.c

```diff
@@ -0,0 +1,258 @@
+	da1469x_qspi_set_bus_mode(QSPIC2_ID, QSPI_BUS_MODE_QUAD);
+
+	reg = QSPIC2->QSPIC2_CTRLMODE_REG;
+	QSPIC2_CTRLMODE_REG_SET_FIELD(QSPIC_SRAM_EN, reg,
+						DT_INST_PROP(0, is_ram));
+	QSPIC2_CTRLMODE_REG_SET_FIELD(QSPIC_USE_32BA, reg,
+						DT_INST_PROP(0, addr_size));
+	QSPIC2_CTRLMODE_REG_SET_FIELD(QSPIC_CLK_MD, reg,
+						DT_INST_PROP(0, clock_mode));
+	QSPIC2_CTRLMODE_REG_SET_FIELD(QSPIC_AUTO_MD, reg, 1);
+	QSPIC2->QSPIC2_CTRLMODE_REG = reg;
+
+	reg = QSPIC2->QSPIC2_BURSTCMDA_REG;
+	QSPIC2_BURSTCMDA_REG_SET_FIELD(QSPIC_DMY_TX_MD, reg,
+						DT_INST_PROP(0, rx_dummy_mode));
+	QSPIC2_BURSTCMDA_REG_SET_FIELD(QSPIC_ADR_TX_MD, reg,
+						DT_INST_PROP(0, rx_addr_mode));
+	QSPIC2_BURSTCMDA_REG_SET_FIELD(QSPIC_INST_TX_MD, reg,
+						DT_INST_PROP(0, rx_inst_mode));
+	QSPIC2_BURSTCMDA_REG_SET_FIELD(QSPIC_EXT_TX_MD, reg,
+						DT_INST_PROP(0, rx_extra_mode));
+	QSPIC2_BURSTCMDA_REG_SET_FIELD(QSPIC_INST, reg,
+						DT_INST_PROP(0, read_cmd));
+	QSPIC2_BURSTCMDA_REG_SET_FIELD(QSPIC_EXT_BYTE, reg,
+						DT_INST_PROP(0, extra_byte));
+	QSPIC2->QSPIC2_BURSTCMDA_REG = reg;
+
+	reg = QSPIC2->QSPIC2_BURSTCMDB_REG;
+	QSPIC2_BURSTCMDB_REG_SET_FIELD(QSPIC_DMY_NUM, reg,
+						DT_INST_PROP(0, dummy_bytes_size));
+	QSPIC2_BURSTCMDB_REG_SET_FIELD(QSPIC_DAT_RX_MD, reg,
+						DT_INST_PROP(0, rx_data_mode));
+	QSPIC2_BURSTCMDB_REG_SET_FIELD(QSPIC_INST_MD, reg, 0);
+	QSPIC2_BURSTCMDB_REG_SET_FIELD(QSPIC_EXT_BYTE_EN, reg,
+						DT_INST_PROP(0, extra_byte_enable));
+	QSPIC2->QSPIC2_BURSTCMDB_REG = reg;
+
+	reg = QSPIC2->QSPIC2_AWRITECMD_REG;
+	QSPIC2_AWRITECMD_REG_SET_FIELD(QSPIC_WR_DAT_TX_MD, reg,
+						DT_INST_PROP(0, tx_data_mode));
+	QSPIC2_AWRITECMD_REG_SET_FIELD(QSPIC_WR_ADR_TX_MD, reg,
+						DT_INST_PROP(0, tx_addr_mode));
+	QSPIC2_AWRITECMD_REG_SET_FIELD(QSPIC_WR_INST_TX_MD, reg,
+						DT_INST_PROP(0, tx_inst_mode));
+	QSPIC2_AWRITECMD_REG_SET_FIELD(QSPIC_WR_INST, reg,
+						DT_INST_PROP(0, write_cmd));
+	QSPIC2->QSPIC2_AWRITECMD_REG = reg;
+}
+
+/* Read PSRAM/NOR device ID using JEDEC commands. */
+static bool memc_jedec_read_and_verify_id(QSPIC_TYPE qspi_id)
+{
+	uint16_t device_density;
+	bool ret = 0;
+	qspi_memory_id_t memory_id;
+
+	da1469x_qspi_memory_jedec_read_id(qspi_id, &memory_id);
+
+	device_density = DT_INST_PROP(0, dev_density);
+	ret |= !(memory_id.id == DT_INST_PROP(0, dev_id));
+	ret |= !(memory_id.type == DT_INST_PROP(0, dev_type));
+	ret |= !((memory_id.density & (device_density >> 8)) == (device_density & 0xFF));
+
+	return ret;
+}
+
+static int memc_smartbond_init(const struct device *dev)
+{
+	uint32_t qspic_ctrlmode_reg;
+
+	/* First QSPI controller is enabled so registers can be accessed */
+	memc_set_status(true, DT_INST_PROP(0, clock_div));
+
+	/* Apply the min. required settings before performing any transaction in manual mode. */
+	qspic_ctrlmode_reg = QSPIC2->QSPIC2_CTRLMODE_REG;
+	QSPIC2_CTRLMODE_REG_SET_FIELD(QSPIC_CLK_MD, qspic_ctrlmode_reg,
+			DT_INST_PROP(0, clock_mode));
```

Comment:
The reason I decided to make this a single-instance driver is because I cannot think of any use case that requires multiple instances; meaning that only a single memory device can be connected physically to the SoC.